java-desktop/src/main/java/com/couchbase/todo/model/DB.java

```diff
@@ -246,25 +247,33 @@ private void startReplication(@NotNull String username, @NotNull String password
         final URI sgUri = getReplicationUri();
         if (sgUri == null) { return; }
 
+        final int attempts = TodoApp.getConfig().getAttempts();
+        final int waitTime = TodoApp.getConfig().getAttemptsWaitTime();
+
         final Endpoint endpoint = new URLEndpoint(sgUri);
 
         final ReplicatorConfiguration config = new ReplicatorConfiguration(db, endpoint)
             .setReplicatorType(ReplicatorConfiguration.ReplicatorType.PUSH_AND_PULL)
-            .setContinuous(true);
+            .setContinuous(true)
+            .setMaxAttempts(attempts)
+            .setMaxAttemptWaitTime(waitTime);
 
         TodoApp.CR_MODE crmode = TodoApp.SYNC_CR_MODE;
         if (crmode == TodoApp.CR_MODE.DEFAULT) {
             config.setConflictResolver(null);
-        } else {
+        }
+        else {
             config.setConflictResolver(conflict -> {
                 Document local = conflict.getLocalDocument();
                 Document remote = conflict.getRemoteDocument();
                 if (local == null || remote == null) { return null; }
                 if (crmode == TodoApp.CR_MODE.LOCAL) {
```

Comment:
Love the improved formatting.  These lines are so short, you could even put them on the same like, as on line 269